bindings/muen/muen-net.c

```diff
@@ -31,37 +32,57 @@ struct net_msg {
     uint16_t length;
 } __attribute__((packed));
 
-static struct muchannel *net_in;
-static struct muchannel *net_out;
-static struct muchannel_reader net_rdr;
+struct muen_net_device {
+    bool acquired;
+    uint8_t mac_addr[6];
+    struct muchannel *net_in;
+    struct muchannel *net_out;
+    struct muchannel_reader net_rdr;
+};
 
-static uint8_t mac_addr[6];
+extern struct mft *muen_manifest;
 
-solo5_result_t solo5_net_write(const uint8_t *buf, size_t size)
+static struct muen_net_device devices[MFT_MAX_ENTRIES];
+
+/**
+ * Initialize Muen network device with given name.
+ */
+static bool muen_net_dev_init(const char *name, struct muen_net_device *device,
+        struct solo5_net_info *info);
+
+solo5_result_t solo5_net_write(solo5_handle_t handle,
+        const uint8_t *buf, size_t size)
 {
     struct net_msg pkt;
 
+    if (handle >= MFT_MAX_ENTRIES || !devices[handle].acquired)
+        return SOLO5_R_EINVAL;
+
     if (size > PACKET_SIZE)
         return SOLO5_R_EINVAL;
 
     memset(&pkt, 0, sizeof(struct net_msg));
     cc_barrier();
     pkt.length = size;
     memcpy(&pkt.data, buf, size);
-    muen_channel_write(net_out, &pkt);
+    muen_channel_write(devices[handle].net_out, &pkt);
 
     return SOLO5_R_OK;
 }
 
-solo5_result_t solo5_net_read(uint8_t *buf, size_t size, size_t *read_size)
+solo5_result_t solo5_net_read(solo5_handle_t handle __attribute__((unused)),
```

Comment:
the `handle` is not `unused` anymore AFAICT

Reply:
I would have expected a compiler warning here... Thanks for pointing it out.